redbeams-api/src/main/java/com/sequenceiq/redbeams/validation/ConnectorJarUrlForDatabaseVendorValidator.java

```diff
@@ -0,0 +1,61 @@
+package com.sequenceiq.redbeams.validation;
+
+import java.util.EnumSet;
+import java.util.Optional;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.DatabaseVendor;
+import com.sequenceiq.cloudbreak.validation.ValidatorUtil;
+import com.sequenceiq.redbeams.api.endpoint.v4.database.request.DatabaseV4Request;
+import com.sequenceiq.redbeams.api.endpoint.v4.databaseserver.requests.DatabaseServerV4Request;
+import com.sequenceiq.redbeams.api.util.DatabaseVendorUtil;
+
+/**
+ * Validates whether the connector JAR URL for a request is valid based on its
+ * database vendor.
+ */
+public class ConnectorJarUrlForDatabaseVendorValidator implements ConstraintValidator<ValidConnectorJarUrlForDatabaseVendor, Object> {
+
+    // Connector JAR URL is optional for PostgreSQL only.
+    private static final EnumSet<DatabaseVendor> DATABASE_VENDORS_ALLOWING_NULL_CONNECTOR_JAR_URL =
+        EnumSet.of(DatabaseVendor.POSTGRES);
+
+    private DatabaseVendorUtil databaseVendorUtil = new DatabaseVendorUtil();
+
+    @Override
+    public boolean isValid(Object request, ConstraintValidatorContext context) {
```

Comment:
This validation needs to apply to multiple data types. `ManagementPackResourceValidator` applies to only one type (`ManagementPackV4Request`), so I can't do the same thing here. Are there any other options?

Reply:
I'm going to leave this as is so that the validation can at least get in place. @topolyai5 If you do think of another option, I'm happy to make changes in a separate review. Thanks!